contracts/apps/accounting/AccountingApp.sol

```diff
@@ -0,0 +1,133 @@
+        // To optimize, incoming transactions could go directly to "Suceeded" or "Failed".
+        updateTransaction(tid, initialState, "new");
+        return tid;
+    }
+
+
+    bytes4 constant UPDATE_TRANSACTION_SIG = bytes4(sha3('updateTransaction(uint transactionId, TransactionState state, string reason)'));
+    function updateTransaction(uint transactionId, TransactionState state, string reason) returns (uint) {
+        uint tuid = transactionUpdates.push(TransactionUpdate({
+            transactionId: transactionId,
+            state: state,
+            reason: reason,
+            actor: msg.sender
+        })) - 1;
+        transactionUpdatesRelation[transactionId].push(tuid);
+        return tuid;
+    }
+
+    bytes4 constant SET_TRANSACTION_SUCCEEDED_SIG = bytes4(sha3('setTransactionSucceeded(uint transactionId, string reason)'));
+    function setTransactionSucceeded(uint transactionId, string reason) {
+        updateTransaction(transactionId, TransactionState.Succeeded, reason);
+    }    
+
+    bytes4 constant SET_TRANSACTION_PENDING_APPROVAL_SIG = bytes4(sha3('setTransactionPendingApproval(uint transactionId, string reason)'));
+    function setTransactionPendingApproval(uint transactionId, string reason) {
+        updateTransaction(transactionId, TransactionState.PendingApproval, reason);
+    }
+
+    bytes4 constant SET_TRANSACTION_FAILED_SIG = bytes4(sha3('setTransactionFailed(uint transactionId, string reason)'));
+    function setTransactionFailed(uint transactionId, string reason) {
+        updateTransaction(transactionId, TransactionState.Failed, reason);
+    }
+
+    // This flattens the last TransactionUpdate with the base Transation to show the current state of the transaction.
+    // This assumes that there is at least a single transaction update which is fine if newTransaction is used.
+    bytes4 constant GET_TRANSACTION_STATE_SIG = bytes4(sha3('getTransactionState(uint transactionId)'));
+    function getTransactionState(uint transactionId) constant returns (address token, int value, string reference, uint timestamp, string stringState, TransactionState state) {
+        Transaction t = transactions[transactionId];
+        uint lastTransactionUpdate = transactionUpdatesRelation[transactionId][transactionUpdatesRelation[transactionId].length - 1];
+        TransactionUpdate tu = transactionUpdates[lastTransactionUpdate];
+        token = t.token;
+        value = t.value;
+        reference = t.reference;
+        timestamp = t.timestamp;
+        state = tu.state;
+        if (state == TransactionState.Succeeded) {
+            stringState = "Succeeded";
+        } else if (state == TransactionState.New) {
+            stringState = "New";
+        } else if (state == TransactionState.PendingApproval) {
+            stringState = "PendingApproval";
+        } else if (state == TransactionState.Failed) 
+            stringState = "Failed";
+    }
+
+    function AccountingApp() {
```

Comment:
If you don't initialize Application with the DAO address, this contract won't produce bytecode as it is abstract. 

Example https://github.com/aragon/aragon-core/blob/master/contracts/apps/bylaws/BylawsApp.sol#L58